packages/plugin-kafka/src/kafka.ts

```diff
@@ -0,0 +1,87 @@
+import assert = require('assert')
+import { resolve } from 'path'
+import { NotFoundError } from 'common-errors'
+import { Microfleet, PluginTypes, LoggerPlugin, PluginInterface } from '@microfleet/core'
+// import retry = require('bluebird-retry')
+import * as Kafka from 'node-rdkafka'
+
+/**
+ * Relative priority inside the same plugin group type
+ */
+export const priority = 0
+export const name = 'kafka'
+export const type = PluginTypes.transport
+
+export type TopicConfig = {
+  topicName: string;
+}
+
+type GlobalConfig = {
+  'metadata.broker.list': string;
+}
+
+type Config = {
+  globalConf: GlobalConfig;
+  topicConf?: TopicConfig;
+}
+
+/**
+ * Defines service extension
+ */
+export interface KafkaPlugin {
+  globalConf: GlobalConfig
+  createConsumer: (topicConf?: TopicConfig) => Kafka.KafkaConsumer
+  createProducer: (topicConf?: TopicConfig) => Kafka.Producer
+}
+
+export class KafkaFactory implements KafkaPlugin {
+  globalConf: GlobalConfig
+  constructor(globalConf: GlobalConfig) {
+    this.globalConf = globalConf
+  }
+
+  createConsumer(topicConf?: TopicConfig): Kafka.KafkaConsumer {
+    return new Kafka.KafkaConsumer(this.globalConf, topicConf)
+  }
+
+  createProducer(topicConf?: TopicConfig): Kafka.Producer {
+    return new Kafka.Producer(this.globalConf, topicConf)
+  }
+}
+
+/**
+ * Defines closure
+ */
+const startupHandlers = (
+  service: Microfleet,
+  globalConf: GlobalConfig
+): PluginInterface => ({
```

Comment:
use something similar to https://github.com/microfleet/core/blob/master/packages/core/src/plugins/logger.ts

as we don't really connect/stop with kafka at this point - we need to only init the plugin and its best done via sync start during plugin init. consult a similar plugin with logger